broker/lib/jobs/BnRStatusPollerJob.js

```diff
@@ -126,7 +134,104 @@ class BnRStatusPollerJob extends BaseJob {
         Promise.resolve(operationStatusResponse)
       );
   }
+
+  static checkOperationCompletionStatus20(job_data) {
+    const operationName = job_data.operation;
+    const instanceInfo = job_data.operation_details;
+    const instance_guid = instanceInfo.instance_guid;
+    const backup_guid = instanceInfo.backup_guid;
+    const deployment = instanceInfo.deployment;
+    const plan = catalog.getPlan(instanceInfo.plan_id);
+    return Promise.try(() => {
+        if (operationName === 'backup') {
+          return DirectorManager
+            .load(plan)
+            .then(directorManager => directorManager.getServiceFabrikOperationState('backup', instanceInfo));
+        }
+      })
+      .tap(operationStatusResponse => {
+        return eventmesh
+          .server
+          .updateAnnotationKey({
+            resourceId: instanceInfo.instance_guid,
+            annotationName: 'backup',
+            annotationType: 'default',
+            annotationId: instanceInfo.backup_guid,
+            key: 'progress',
+            value: JSON.stringify(operationStatusResponse)
+          });
+      })
+      .then(operationStatusResponse => {
+        operationStatusResponse.jobCancelled = false;
+        operationStatusResponse.operationTimedOut = false;
+        operationStatusResponse.operationFinished = false;
+        if (utils.isServiceFabrikOperationFinished(operationStatusResponse.state)) {
+          operationStatusResponse.operationFinished = true;
+          return operationStatusResponse;
+        } else {
+          logger.info(`Instance ${instance_guid} ${operationName} for backup guid ${backup_guid} still in-progress - `, operationStatusResponse);
+          const currentTime = new Date();
+          const backup_triggered_duration = (currentTime - new Date(instanceInfo.started_at)) / 1000;
+          return Promise
+            .try(() => bosh.director.getDirectorConfig(instanceInfo.deployment))
+            .then(directorConfig => {
+              const lock_deployment_max_duration = directorConfig.lock_deployment_max_duration;
+              if (backup_triggered_duration > lock_deployment_max_duration) {
+                //Operation timed out
+                if (!instanceInfo.abortStartTime) {
+                  //Operation not aborted. Aborting operation and with abort start time
+                  // re-registering statupoller job
+                  let abortStartTime = new Date().toISOString();
+                  instanceInfo.abortStartTime = abortStartTime;
+                  return DirectorManager
+                    .load(plan)
+                    .then(directorManager => directorManager.abortLastBackup(instanceInfo.tenant_id,
```

Comment:
Abort should happen via 2.0 implementation. `directorManager.abortLastBackup`  is 1.0

Reply:
Done